src/option_learning.py

```diff
@@ -212,3 +220,206 @@ def update_segment_from_option_spec(self, segment: Segment,
                                   dtype=np.float32)
                 option = param_opt.ground([robby, block], params)
                 segment.set_option(option)
+
+
+@dataclass(frozen=True, eq=False, repr=False)
+class _LearnedSimpleParameterizedOption(ParameterizedOption):
+    """A convenience class for holding a learned parameterized option.
+
+    Prefer to use this because it is pickleable.
+    """
+    _operator: STRIPSOperator
+    _regressor: MLPRegressor
+    _changing_parameter_idxs: Sequence[int]
+
+    def __init__(self, name: str, operator: STRIPSOperator,
+                 regressor: MLPRegressor,
+                 changing_parameters: Sequence[Variable]) -> None:
+        assert set(changing_parameters).issubset(set(operator.parameters))
+        changing_parameter_idxs = [
+            i for i, v in enumerate(operator.parameters)
+            if v in changing_parameters
+        ]
+        types = [v.type for v in operator.parameters]
+        option_param_dim = sum(v.type.dim for v in changing_parameters)
+        params_space = Box(low=-np.inf,
+                           high=np.inf,
+                           shape=(option_param_dim, ),
+                           dtype=np.float32)
+        # Dataclass is frozen, so we have to do these hacks.
+        object.__setattr__(self, "_operator", operator)
+        object.__setattr__(self, "_regressor", regressor)
+        object.__setattr__(self, "_changing_parameter_idxs",
+                           changing_parameter_idxs)
+        super().__init__(name,
+                         types,
+                         params_space,
+                         _policy=self._regressor_based_policy,
+                         _initiable=self._precondition_based_initiable,
+                         _terminal=self._effect_based_terminal)
+
+    @cached_property
```

Comment:
Note: you have to be careful with cached_property if you're expecting that the self._operator might change at any point

Reply:
the operator shouldn't change